### PR TITLE
Stage 1: Main Code

### PR DESCRIPTION
**Pull request by sm86**
Re attempted to replicate Stage 1 of paper.
Input parameters: Wiki Dumps
Output: Leaves inserted in NODE table
and Single level hierarchy in EDGE table

looking for your suggestions/comments on
my coding style and anyways to improve/anything to focus on.

### CODE REVIEW

File: original/java/db.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#mysql DB properties
```
Comment (jimkont) [inline original/java/db.properties]: better practice to have this as e.g. `db.properties.default` that people will copy to `db.properties` to run the code. this way there will be no code conflicts as people use the code

File: original/java/src/main/java/org/dbpedia/categories/database/DatabaseConnection.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			// load the Driver Class
+			Class.forName("com.mysql.jdbc.Driver").newInstance();
+
+			// create the connection now	
+			con = DriverManager.getConnection(props.getProperty("DB_URL"),
+					props.getProperty("DB_USERNAME"),
+					props.getProperty("DB_PASSWORD"));
+
+		} catch (IOException | ClassNotFoundException | SQLException | InstantiationException | IllegalAccessException e) {
+			// TODO Auto-generated catch block
```
Comment (jimkont) [inline original/java/src/main/java/org/dbpedia/categories/database/DatabaseConnection.java]: logging is a better option here, for now it is OK but you should mark it as a todo for later. Log4j2 / slf4j is a good option

File: original/java/src/main/java/org/dbpedia/categories/database/EdgeDB.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+			ps.setInt(1, parentId);
+			ps.setInt(2, chidId);
+			ps.executeUpdate();
+			connection.close();
+		}catch(SQLException e){
+			e.printStackTrace();
+		}
+	}
+	/*
+
```
Comment (jimkont) [inline original/java/src/main/java/org/dbpedia/categories/database/EdgeDB.java]: dead code should be avoided as much as possible

File: original/java/src/test/java/PageDBTest.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import org.dbpedia.categories.database.PageDB;
+
+public class PageDBTest {
+
+	public static void main(String[] argv){
```
Comment (jimkont) [inline original/java/src/test/java/PageDBTest.java]: can you add some explanation on what this code does? it is on the test folder but not very clear.

File: original/java/db.properties (hunk 1 above)
Comment (sm86) [reply original/java/db.properties]: I think I missed something. Do you mean to change the file name to db.properties.default?

File: original/java/src/main/java/org/dbpedia/categories/database/DatabaseConnection.java (hunk 2 above)
Comment (sm86) [reply original/java/src/main/java/org/dbpedia/categories/database/DatabaseConnection.java]: Opened an issue #4 
Will work on it.
Thanks!

File: original/java/src/main/java/org/dbpedia/categories/database/EdgeDB.java (hunk 3 above)
Comment (sm86) [reply original/java/src/main/java/org/dbpedia/categories/database/EdgeDB.java]: Sure, will take care on this point from next PRs. Thanks.

File: original/java/src/test/java/PageDBTest.java (hunk 4 above)
Comment (sm86) [reply original/java/src/test/java/PageDBTest.java]: I am reading about Unit Testing and will complete this part by next commit in a day or two.

Comment (sm86) [reply original/java/src/test/java/PageDBTest.java]: Maybe I should concentrate on this part after I make the workflow working. I just found that Eclipse IDE has few methods to generate these. 
I will remove this file and make a commit.

**Pull request closed without merging**